lib/cc/engine/analyzers/reporter.rb

```diff
@@ -1,22 +1,45 @@
 require 'cc/engine/analyzers/violation'
+require 'thread'
 
 module CC
   module Engine
     module Analyzers
       class Reporter
         TIMEOUT = 10
-        def initialize(language_strategy, io)
+
+        def initialize(config, language_strategy, io)
+          @config = config
           @language_strategy = language_strategy
           @io = io
         end
 
         def run
-          sexps = language_strategy.run
+          queue = Queue.new
 
-          sexps.each do |sexp|
-            process_sexp(sexp)
+          language_strategy.each_file do |file|
+            queue.push(file)
           end
 
+          workers = @config.concurrency.times.map do
+            Thread.new do
+              begin
+                while file = queue.pop(true)
+                  begin
```

Comment:
Could the begin/rescue/end for processing files also be extracted into another method?